### PR TITLE
Implement :run-in-background option

### LINKED ISSUE #3: Abilility to run in background
I was thinking that it'd be great to e.g. `lein do nrepl :in-background, some-other-long-running-task`.

That way I can use the same JVM process to run a nrepl server, and concurrently another long-running task. Saves resources basically.

Would it be a welcome feature? As conceived lein-nrepl seems a temporary library, but perhaps it can aspire to a bit more.

Can give a shot to a PR if the idea seems sound.

Cheers - Victor

### PR DESCRIPTION
**Pull request by vemv**
Fixes #3 

How I tested it out:

* Install https://github.com/gfredericks/lein-lein/
* Run `lein do nrepl :port 42111 :run-in-background 1, echo "Foo", hang`
* See that `Foo` is echoed, and that `hang` runs indefinitely while I still can access port 42111

Cheers - V

### CODE REVIEW

**Comment by bbatsov** [comment]
How are you supposed to use this without something like `hang`, btw?

**Comment by vemv** [comment]
https://github.com/jakemcc/lein-test-refresh would be a good example of an indefinitely-running task. Same for cljsbuild or such

**Comment by vemv** [comment]
Btw a better name for the option could be `:do-not-block`, otherwise the behavior can be surprising when running `lein nrepl :run-in-background true` (user might have expected sth like a daemonized process, he got a terminated process instead).

Can change after code review.

**Comment by bbatsov** [comment]
The code changes look good to me, I'm simply concerned about the name. `:run-in-background` is certainly confusing. I don't like names with `not` in them - guess it can be `:block false` instead.

File: src/leiningen/nrepl.clj [hunk 1]
```diff
@@ -34,14 +34,19 @@
      resolve to a sequence of vars, in which case they'll be flattened into the
      list of middleware.
 
+   * :run-in-background - Set this for relinquishing control to the next Leiningen task:
```
Comment (bbatsov) [inline src/leiningen/nrepl.clj]: I'd also mention this in the README.

**Comment by vemv** [comment]
Amended!

**Pull request merged**